### DISCUSSION TITLE
Microdot-websocket-asyncio connection for ESP32-CAM

### DISCUSSION THREAD

**Discussion by qqjyh** [miguelgrinberg/microdot#257] [Q&A]
Microdot-websocket-asyncio has superior performance and is easy to use, but I cannot find any information or examples about connecting to ESP32CAM ?

**Comment by miguelgrinberg** [comment]
Microdot only provides the websocket server. To connect to the server you will need to use a websocket client. There are literally hundreds of clients available for every possible platform, programming language and operating system. You will need to look for connection examples in your chosen client.

**Reply by qqjyh** [reply]
Thanks for repid reply! The websocket server that you meat Microdot provides is actually realized by Esp32-cam. So the micropython program examples running on Esp32-cam should be provided, which use buf=camera.capture() and the buf need treat into the correct format to be sent by the websocket server. The client side I use http-js, no problem !

**Reply by qqjyh** [reply]
I would like to ask another question: From the several original mPython. py program of  microdot-websockets, it seems that there is a websocket-client interface (I am still learning Python programming). Can microdot-websocket also be used as a websocket-client?

**Reply by miguelgrinberg** [reply]
The WebSocket server provided by Microdot is completely generic, and I prefer to show examples that are also generic and not dependent on a specific board. Unfortunately there are a large number of different boards, so it would be an impossible task to provide examples that are specific to each!
 
And as I said above, Microdot is a server, and as such it only provides a WebSocket server component. If you need to use a client inside your device, then you will need to look for a client component elsewhere.

**Reply by qqjyh** [reply]
Thank you very much !
